miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultBoxService.java

```diff
@@ -264,13 +277,44 @@ public long save(Box box) throws IOException {
     }
   }
 
+  /**
+   * Checks submitted data for validity, throwing a ValidationException containing all of the errors if invalid
+   * 
+   * @param box submitted Box to validate
+   * @param beforeChange the already-persisted Box before changes
+   * @throws IOException
+   */
+  private void validateChange(Box box, Box beforeChange) throws IOException {
+    List<ValidationError> errors = new ArrayList<>();
+
+    if (box.getStorageLocation() != null) {
+      if (box.getStorageLocation().getLocationUnit().getBoxStorageAmount() == BoxStorageAmount.NONE) {
+        errors.add(new ValidationError("storageLocation", "Invalid box location"));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)